controllers/networkfence_controller.go

```diff
@@ -152,92 +191,101 @@ func (r *NetworkFenceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-// fenceClusterNetwork adds a finalizer and sends a FenceClusterNetwork request.
-func (r *NetworkFenceReconciler) fenceClusterNetwork(ctx context.Context, nwFence *csiaddonsv1alpha1.NetworkFence, controllerClient proto.NetworkFenceClient, logger logr.Logger) error {
+// processFencing adds a finalizer and handles the fencing request.
+func (nf *NetworkFenceInstance) processFencing(ctx context.Context) error {
 
 	// add finalizer to the networkfence object if not already present.
-	if err := r.addFinalizerToNetworkFence(ctx, &logger, nwFence); err != nil {
-		logger.Error(err, "Failed to add NetworkFence finalizer")
+	if err := nf.addFinalizerToNetworkFence(ctx); err != nil {
+		nf.logger.Error(err, "Failed to add NetworkFence finalizer")
 		return err
 	}
 
+	return nf.processFencingRequest(ctx)
+}
+
+// processFencingRequest creates the fencing request based on
+// the spec and then calls appropriate function to either
+// fence or unfence based on the spec.
+func (nf *NetworkFenceInstance) processFencingRequest(ctx context.Context) error {
 	// send FenceClusterNetwork request.
-	logger.Info("FenceClusterNetwork Request")
-	timeoutContext, cancel := context.WithTimeout(ctx, r.Timeout)
+	if nf.instance.Spec.FenceState == csiaddonsv1alpha1.Fenced {
+		nf.logger.Info("FenceClusterNetwork Request")
+	} else {
+		nf.logger.Info("UnFenceClusterNetwork Request")
+	}
```

Comment:
this is not helpful logging. log in at Reconcile function with namespaced name.

Reply:
ACK.